src/picongpu/include/plugins/particleCalorimeter/ParticleCalorimeter.hpp

```diff
@@ -0,0 +1,530 @@
+private:
+    typedef boost::shared_ptr<MyCalorimeterFunctor> MyCalorimeterFunctorPtr;
+    MyCalorimeterFunctorPtr calorimeterFunctor;
+
+    typedef boost::shared_ptr<PMacc::algorithm::mpi::Reduce<simDim> > AllGPU_reduce;
+    AllGPU_reduce allGPU_reduce;
+
+    void restart(uint32_t restartStep, const std::string restartDirectory)
+    {
+        if(this->notifyPeriod == 0)
+            return;
+
+        HBufCalorimeter hBufLeftParsCalorimeter(this->dBufLeftParsCalorimeter->size());
+
+        int rank;
+        MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+        if(rank == 0)
+        {
+            splash::SerialDataCollector hdf5DataFile(1);
+            splash::DataCollector::FileCreationAttr fAttr;
+
+            splash::DataCollector::initFileCreationAttr(fAttr);
+            fAttr.fileAccType = splash::DataCollector::FAT_READ;
+
+            std::stringstream filename;
+            filename << restartDirectory << "/" << prefix << restartStep;
+
+            hdf5DataFile.open(filename.str().c_str(), fAttr);
+
+            splash::Dimensions dimensions;
+
+            hdf5DataFile.read(restartStep,
+                              this->leftParticlesDatasetName.c_str(),
+                              dimensions,
+                              &(*hBufLeftParsCalorimeter.origin()));
+
+            hdf5DataFile.close();
+
+            /* rank 0 divides and distributes the calorimeter to all ranks in equal parts */
+            int numRanks;
+            MPI_Comm_size(MPI_COMM_WORLD, &numRanks);
+            using namespace lambda;
+            PMacc::algorithm::host::Foreach()(hBufLeftParsCalorimeter.zone(),
+                                              hBufLeftParsCalorimeter.origin(),
+                                              _1 = _1 / float_X(numRanks));
+        }
+
+        MPI_Bcast(&(*hBufLeftParsCalorimeter.origin()),
+                  hBufLeftParsCalorimeter.size().productOfComponents() * sizeof(float_X),
+                  MPI_CHAR,
+                  0,
+                  MPI_COMM_WORLD);
+
+        *this->dBufLeftParsCalorimeter = hBufLeftParsCalorimeter;
+    }
+
+
+    void checkpoint(uint32_t currentStep, const std::string checkpointDirectory)
+    {
+        if(this->notifyPeriod == 0)
+            return;
+
+        HBufCalorimeter hBufLeftParsCalorimeter(this->dBufLeftParsCalorimeter->size());
+        HBufCalorimeter hBufTotal(hBufLeftParsCalorimeter.size());
+
+        hBufLeftParsCalorimeter = *this->dBufLeftParsCalorimeter;
+
+        /* mpi reduce */
+        using namespace lambda;
+        (*this->allGPU_reduce)(hBufTotal, hBufLeftParsCalorimeter, _1 + _2);
+        if(!this->allGPU_reduce->root())
+            return;
+
+        splash::SerialDataCollector hdf5DataFile(1);
+        splash::DataCollector::FileCreationAttr fAttr;
+
+        splash::DataCollector::initFileCreationAttr(fAttr);
+
+        std::stringstream filename;
+        filename << checkpointDirectory << "/" << prefix << currentStep;
```

Comment:
pls give our checkpoints a suffix with `.<timeStep>` (example for [txt files](https://github.com/ComputationalRadiationPhysics/picongpu/blob/f5fe0487f8ed070894feb11a40da11703a9f2391/src/picongpu/include/plugins/common/txtFileHandling.hpp#L88-L112))

suggested change:
```C++
filename << checkpointDirectory << "/" << prefix << "." << currentStep;
```

for `checkpoints/<species>_calorimeter_0_0_0.h5.<step>`